src/main/java/com/hotels/beeju/core/BeejuCore.java

```diff
@@ -75,11 +87,27 @@ public BeejuCore(String databaseName, Map<String, String> configuration) {
       // overriding default derby log path to go to tmp
       String derbyLog = File.createTempFile("derby", ".log").getCanonicalPath();
       System.setProperty("derby.stream.error.file", derbyLog);
+
+      //Creating temporary folder
+      init();
```

Comment:
I think either have an external class call `init()` and don't init in the constructor or leave the `init()` call here and make the method private? Nothing is overwriting it atm so it can be private

Reply:
if private also please rename to something like: `createWarehousePath()`

Reply:
yeah i'll make this private and change the name, this part should happen during core initialisation to have access to the temporary directory. Although it is only 2 lines, which could be added to the constructor instead?